### PR TITLE
Remove fixed size of EmptyState button

### LINKED ISSUE #269: `EmptyState` button has fixed vertical size
The `EmptyState` now uses `fixedSize` `Button` to keep the width as needed. 

This works for small content, but makes the button grow over provided width. Applying `.fixedSize` on stack inside the `Button` does not work, as it makes the whole container fixedSize and does not respect the `Text` inside being "not" fixedSize (in horizontal axis).

#### Solution
The horizontal (and vertical) expanding property of all components will need to be configurable (disabled) by a modifier.

#### Suggestions
- modifier `.idealSize(horizontal:vertical)` (similar to `.fixedSize()`) that propagates information to subviews that they should not be expanding beyond the ideal size in the provided axis.

### PR DESCRIPTION
**Pull request by PavelHolec**
Closes #269 

Applied the modifier to other expanding components.

image

image

image

image

### CODE REVIEW

**Review by Parabak — APPROVED** [review]

**Comment by PavelHolec** [comment]
@sjavora @Parabak please recheck - the only solution I found to address the issue is to add a new modifier (see issue description)

File: Sources/Orbit/Components/Button.swift [hunk 1]
```diff
@@ -289,21 +293,26 @@ struct ButtonPreviews: PreviewProvider {
...
+            }
+            .idealSize(horizontal: true)
+
+//            standaloneCombinations
+//            sizing
+//
+//            storybook
+//            storybookStatus
+//            storybookGradient
+//            storybookMix
```
Comment (sjavora) [inline Sources/Orbit/Components/Button.swift:308]: Commented code.

File: Sources/Orbit/Components/Button.swift [hunk 2]
```diff
@@ -432,32 +441,32 @@ struct ButtonPreviews: PreviewProvider {
...
+//        .padding(.medium)
+//        .previewLayout(.sizeThatFits)
+//    }
+//
+//    @ViewBuilder static var content: some View {
+//        ButtonPreviews.standaloneCombinations
+//        ButtonPreviews.sizing
+//        ButtonPreviews.buttons(.primary)
+//    }
+//}
```
Comment (sjavora) [inline Sources/Orbit/Components/Button.swift:472]: More commented code.

**Review by sjavora — CHANGES_REQUESTED** [review]

File: Sources/Orbit/Components/Button.swift (hunk 2 above)
Comment (PavelHolec) [reply Sources/Orbit/Components/Button.swift]: Thx, updated both

**Review by sjavora — APPROVED** [review]

**Pull request merged**